jobs/config.json

```diff
@@ -12306,6 +12306,18 @@
       "sig-apps"
     ]
   },
+  "pull-cloud-provider-azure-e2e": {
+    "args": [
+      "make",
```

Comment:
I think this will do as a start, but we might want to bring it a bit more in line with how the other e2e tests are run in CI.